File: modules/angular2/src/core/compiler/directive_lifecycle_reflector.ts
import {Type} from 'angular2/src/core/facade/lang';
import * as Interfaces from './interfaces';

export function hasLifecycleHook(lcInterface, type): boolean {
  if (!(type instanceof Type)) return false;

  var proto = (<any>type).prototype;

  switch (lcInterface) {
    case Interfaces.AfterContentInit:
      return !!proto.afterContentInit;
    case Interfaces.AfterContentChecked:
      return !!proto.afterContentChecked;
    case Interfaces.AfterViewInit:
      return !!proto.afterViewInit;
    case Interfaces.AfterViewChecked:
      return !!proto.afterViewChecked;
    case Interfaces.OnChanges:
      return !!proto.onChanges;
    case Interfaces.DoCheck:
      return !!proto.doCheck;
    case Interfaces.OnDestroy:
      return !!proto.onDestroy;
    case Interfaces.OnInit:
      return !!proto.onInit;
    default:
      return false;
  }
}

File: modules/angular2/src/core/compiler/directive_lifecycle_reflector.dart
library angular2.src.core.compiler.directive_lifecycle_reflector;

import 'package:angular2/src/core/reflection/reflection.dart';

bool hasLifecycleHook(/*LifecycleHook*/ interface, type) {
  if (type is! Type) return false;

  return reflector.interfaces(type).contains(interface);
}
